torch/csrc/distributed/rpc/tensorpipe_agent.h

```diff
@@ -159,6 +162,31 @@ class TensorPipeAgent : public RpcAgent {
   mutable std::mutex mutex_;
   uint64_t nextMessageID_{0};
 
+  // Map to store the expiration times for each message.
+  std::map<steady_clock_time_point, std::vector<std::shared_ptr<FutureMessage>>>
+      timeoutMap_;
```

Comment:
Is there a specific reason to use a `std::map<TKey, std::vector<TValue>>` rather than a `std::multimap<TKey, TValue>`? I guess dealing with an entire vector at a time may be more efficient than dealing with multiple futures over multiple iterations, but we also don't really expect to have that many futures with _exactly_ the same timeout, down to the same millisecond. And dealing with one future at a time rather than a whole vector of them would simplify the above code further.